src/cli/glob_base_dir.rs

```diff
@@ -30,25 +31,51 @@
 //!
 //! [`MIT license`]: https://docs.rs/crate/globwalk/0.8.0/source/LICENSE
 
+use anyhow::anyhow;
 use globwalk::GlobWalkerBuilder;
 use std::path::PathBuf;
 
-/// Function to determine the base directory and (remaining) pattern.
+/// Create a generic glob builder
+pub fn glob_builder_base<PAT: AsRef<str>>(
```

Comment:
I realized something here: we can probably combine this implementation with `CommonDir` specified elsewhere (they both have the goal of finding the greatest common base directory; here so the smallest amount of paths have to be matched against, and there to find the top level for mirroring the directory structure). For this PR, that doesn't need to be included however, since it was already separately available prior to this PR.